packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2606,33 +2606,45 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
                 restorationId: 'time_picker_scroll_view_vertical',
                 child: AnimatedContainer(
                   width: allowedSize.width,
-                  height: allowedSize.height,
                   duration: _kDialogSizeAnimationDuration,
                   curve: Curves.easeIn,
+                  constraints: BoxConstraints(
+                    minHeight: _kTimePickerMinInputSize.height,
```

Comment:
This should be the original value. So above we'd want:
```dart
static const Size _kTimePickerInputSize = Size(312, 252);
static const double _kTimePickerInputMinimumHeight = 216;
```

then here it would be:
```dart
constraints: BoxConstraints(
  minHeight: _kTimePickerInputMinimumHeight,
  maxHeight: allowedSize.height,
),
```
This keeps dialogs that do not need to expand to not be bigger than they were before.

Reply:
Sorry, My bad I didn't checked value of _kTimePickerMinInputSize and thought it will be 216. I have pushed required change.